src/Workspaces/Core/Portable/CodeGeneration/CodeGenerationContext.cs

```diff
@@ -4,11 +4,17 @@
 
 using System;
 using System.Collections.Generic;
+using Microsoft.CodeAnalysis.AddImport;
 using Microsoft.CodeAnalysis.Simplification;
 using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.CodeGeneration
 {
+    internal readonly record struct CodeGenerationSolutionContext(
```

Comment:
Groups information needed for generating code to arbitrary file in the solution. This may require generating code in more than one language and thus language specific fallback options.